hw/ip/rom_ctrl/util/gen_vivado_mem_image.py

```diff
@@ -55,14 +217,23 @@ def main() -> int:
 
     # OpenTitan vmem files should always contain one single contiguous chunk.
     assert len(vmem.chunks) == 1
+    words = vmem.chunks[0].words
+
+    if width == 24:
```

Comment:
> nit: `width` is an awkward key for deciding the behavior of this tool. How are the two paths different, and is there a program option / argument we could add so the user can explicitly request what they want?

Agreed. It would be trivial to add a `--mode=(otp|rom)` flag. Or are you asking about trying to unify the two modes and induce the behavior we want with a parameter? That would be a trickier, but not impossible.

For each ROM word, we reverse the nibbles, convert to a hex string, and we're done. OTP splicing needs to reverse the bits, pad with 15*22 zero bits, and emit 44-bit-wide hex strings. So perhaps we could define `--reverse_bits` and `--hex-word-width`. I'd rather not bake more information into the vmem filename, so maybe we need a special `otp_splice` rule in `//rules/splice.bzl`.

> Also nit: This script is in an awkward place, as it is associated with rom_ctrl. With this directory structure, it's a bit funky for otp_ctrl to be using it.
> 
> I think the generalizing of the script can go in another PR, though.

Yep, I agree. I feel like it might make sense to emit both the vmem for `Vivado` and the MEM file for `updatemem` at the same time rather than having this preprocessing step. I'd like to punt on this issue, but I'll file an issue before resolving this.

Reply:
Filed a bug to track generalizing the script: #15552

I'll move the script somewhere under `//hw/bitstream`.